### PR TITLE
[JA] Update /site/ja/guide/function.ipynb

### PR DESCRIPTION
**Pull request by AseiSugiyama**
I updated outdated notebook function.ipynb.
Original document: https://github.com/tensorflow/docs/blob/master/site/en/guide/function.ipynb

See also: https://github.com/tensorflow/docs-l10n/issues/145

### CODE REVIEW

File: site/ja/guide/function.ipynb [hunk 1]
```diff
@@ -486,16 +485,61 @@
...
+        "colab": {},
+        "colab_type": "code",
+        "id": "i1HtHX6Ja4_o"
+      },
+      "outputs": [],
+      "source": [
+        "@tf.function\n",
+        "def f(x):\n",
+        "  if x > 0:\n",
+        "    # Try setting a breakpoint here!\n",
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:515]: Do we need to translate this into Japanese?

File: site/ja/guide/function.ipynb [hunk 2]
```diff
@@ -486,16 +485,61 @@
...
+        "    # Example:\n",
+        "    #   import pdb\n",
+        "    #   pdb.set_trace()\n",
+        "    x = x + 1\n",
+        "  return x\n",
+        "\n",
+        "tf.config.experimental_run_functions_eagerly(True)\n",
+        "\n",
+        "# You can now set breakpoints and run the code in a debugger.\n",
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:524]: Do we need to translate this into Japanese?

File: site/ja/guide/function.ipynb [hunk 3]
```diff
@@ -486,16 +485,61 @@
...
       "metadata": {
         "colab_type": "text",
         "id": "4LfnJjm0Bm0B"
       },
       "source": [
-        "## 例: シンプルなモデルの学習\n",
+        "## 応用例: グラフ内での訓練ループ\n",
+        "\n",
+        "前のセクションでは AutoGraph を Keras のレイヤーやモデルの中で利用できることを見てきました。\n",
+        "逆に、AutoGraph の中で Keras モデルを用いることもできます。\n",
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:540]: nit: "逆に" looks good. But, no English word that corresponds to this.

File: site/ja/guide/function.ipynb [hunk 4]
```diff
@@ -707,6 +751,204 @@
...
+      "metadata": {
+        "colab_type": "text",
+        "id": "JcVw0Gv5a4_9"
+      },
+      "source": [
+        "## 再トレーシング\n",
+        "\n",
+        "要点を次に示します:\n",
+        "\n",
+        "* 関数の引数に Tensor でない変数や、shape を変えるような引数を渡す際には注意が必要\n",
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:766]: [suggestion]
```suggestion
        "* 関数に Tensor でない引数や、shape を変えるような引数を渡す際には注意が必要\n",
```

File: site/ja/guide/function.ipynb [hunk 5]
```diff
@@ -707,6 +751,204 @@
...
+        "* モジュール化された関数や、モジュール化されたクラスのメソッドに対してデコレートし、ローカルの関数やメソッドに対するデコレートは避ける\n",
+        "\n",
+        "`tf.function` は最初の実行時に遅くなるというコストが必要ですが、Eager Exdecution よりも格段に早く実行されます。これは、最初の実行時に関数がトレースされ、TensorFlow のグラフに変換されるためです。最適化されたグラフを構築するのには、それを実行するよりもずっと時間がかかります。"
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:769]: [suggestion]
```suggestion
        "`tf.function` は最初の実行時に遅くなるというコストが必要ですが、Eager Execution よりも格段に早く実行されます。これは、最初の実行時に関数が *トレース* され 、TensorFlow のグラフに変換されるためです。最適化されたグラフを構築するのには、大抵それを実行するよりもずっと時間がかかります。"
```

File: site/ja/guide/function.ipynb [hunk 6]
```diff
@@ -707,6 +751,204 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "pvSTSjvha5AC"
+      },
+      "source": [
+        "Tensor でない異なる値を引数として渡したとき、`tf.function` は**再トレース**を行うことがあります\b。"
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:837]: [weak suggestion]
```suggestion
        "Tensor でない異なる引数として渡したとき、`tf.function` は**再トレース**を行うことがあります\b。"
```

File: site/ja/guide/function.ipynb [hunk 7]
```diff
@@ -707,6 +751,204 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "XL0kyRYYa5AG"
+      },
+      "source": [
+        "**再トレース**は `input_signature` を指定しない場合、 Tensor の shape が異なるものを引数に渡したときも発生します。"
```
Comment (kiszk) [inline site/ja/guide/function.ipynb:869]: [suggstion]
```suggestion
        "*再トレース*は `input_signature` を指定しない場合、 Tensor の shape が異なるものを引数に渡したときも発生します。"
```

File: site/ja/guide/function.ipynb (hunk 1 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:515]: IMO, it is so hard to maintain these comments in the code cell translated into Japanese. I guess, with the code sync tool `nb_code_sync.py`, this comment will be overwritten by the original one when the original English notebook was updated.

I translate these comments into Japanese just now, however, we need to discuss this issue in the Japanese translation community.

File: site/ja/guide/function.ipynb (hunk 2 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:524]: Same as above, I translate this comment.

File: site/ja/guide/function.ipynb (hunk 3 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:540]: I removed it.

File: site/ja/guide/function.ipynb (hunk 4 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:766]: Nice suggestion 👍

File: site/ja/guide/function.ipynb (hunk 5 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:769]: Thanks, I fixed it.

File: site/ja/guide/function.ipynb (hunk 7 above)
Comment (AseiSugiyama) [reply site/ja/guide/function.ipynb:869]: I fixed it.

**Comment by AseiSugiyama** [comment]
Thank you for your review, @kiszk ! I fixed after your review, please check it again.

**Comment by kiszk** [comment]
Thank you, LGTM

**Comment by lamberta** [comment]
Thanks

As far as translating comments, I think it's a nice to have, but does make maintenance more difficult to keep things in sync with the en/ versions. Ideally, import comments are moved into a separate text cell ... but that's difficult to catch.

**Pull request merged**